### ISSUE TITLE
Allow routes named “array”

### ISSUE THREAD

**Issue by jenweber** [emberjs/ember.js#16872]
@rwjblue commented that making “array” a reserved route name is an artifact of things that we don’t do anymore.

Robert, please confirm if my summary is correct and add some details. Feel free to edit my description directly. 

I’m wondering if this is a good issue for new contributors too.

**Comment by rwjblue** [comment]
Yep! That’s totally correct! We no longer need to assert when you do `this.route(‘array’)` in the `Router.map`.
